docs/en/operation/User-CLI.md

```diff
@@ -1302,6 +1302,33 @@ $ ./bin/alluxio fs rm /tmp/unused-file
 $ ./bin/alluxio fs rm --alluxioOnly /tmp/unused-file2
 ```
 
+When deleting only from Alluxio but leaving the files in UFS, you can use `-U` and `-Dalluxio.user.file.metadata.sync.interval=-1`
+to skip the metadata sync and the UFS check. This will save time and memory consumption on the Alluxio master.
+```console
+$ bin/alluxio fs rm -R -U --alluxioOnly -Dalluxio.user.file.metadata.sync.interval=-1 /path
+```
+
+When deleting a large directory (with millions of files) recursively both from Alluxio and UFS,
+the operation is expensive. 
+
+We recommend doing the deletion in the following way:
+1. Using the corresponding tool to check the UFS path directly and make sure everything can be deleted safely. 
+For example if the UFS is HDFS, use `hdfs dfs -ls -R /dir` to list the UFS files and check.
+Please do not sync the directory from Alluxio and check with `alluxio fs ls -R /dir`, because the loaded file metadata will
+be deleted anyway and the expensive metadata sync operation will essentially be wasted.
+2. Then issue the deletion from Alluxio and delete both in-Alluxio and in-UFS files:
+```console
+# Disable the sync and skip the UFS check, to reduce memory consumption on the master side
+$ bin/alluxio fs rm -R -U -Dalluxio.user.file.metadata.sync.interval=-1 /dir
+```
+
+If the metadata sync and UFS check are both disabled, deleting 1 million files from Alluxio will increase the JVM heap
+by around 2.5GB. If the UFS check is disabled by `-U`, deleting the files from UFS as well introduces around 10% extra overhead.
+Alluxio 2.8 introduced improvements to memory usage and reduced the JVM heap increase by around 20%.
```

Comment:
the before and after version comparison belongs in the release notes, but not in documentation. realistically, the docs should never mention "starting from version x" or "y was added in version x".

Reply:
I'd like this make this as obvious as possible because I don't think people read the release note that attentively. Is there a better way?

Reply:
but what are the chances that someone is reading this section as part of upgrading from pre-2.8 to post-2.8 or investigating the difference between as opposed to reading the release notes? there's never a good place to put it in the main docs; the release notes and other release related activities are the best places to point attention to version diffs. if people aren't reading the release notes and relying on searching through the individual doc pages, then we're not doing a good enough job of providing the version specific information.

another angle is what would the docs would look like if we listed all the "in version x, this changed" within the docs?

Reply:
Ok on 2nd thought I will remove this line from here, it belongs better to the release notes.

A digression - I think the user tends to just go to the latest doc page instead of going to the specific version first, so it might actually make sense in the latest version page to leave a pointer somehow.

Reply:
no reader should need to jump back and forth to do the math. list the relevant numbers in both places.

Reply:
Ok removed the version and only 2.8 stats are kept